KeystreamGenerator.rb

```diff
@@ -0,0 +1,138 @@
+class KeystreamGenerator
+
+  attr_reader :keystream_length, :deck
+
+  def initialize(keystream_length)
+    @keystream_length = keystream_length
+  end
+
+  def generate_keystream
+    @deck = build_deck
+    generate_letters
+  end
+
+  private
+
+  NUMBER_CONVERTER = Hash[(1..26).zip ('A'..'Z')]
+
+  def build_deck
+    new_deck = (1..52).to_a
+    new_deck << 'A'
+    new_deck << 'B'
+    new_deck
+  end
+
+
+  def generate_letters
+    new_keystream = []
+    letters_added = 0
+
+    until letters_added == keystream_length do
+      new_keystream << cut_deck_and_output_letter
+      new_keystream.last == nil ? letters_added : letters_added += 1
```

Comment:
Style: I prefer to expand ternary operators. It's less likely to hide conditional logic, which is both harder to follow and expensive, from a complexity perspective. Also, it's less likely you'll remember to write tests for all logical forks.